app/main.py

```diff
@@ -1,2 +1,61 @@
+from operator import index
+from typing import Any
+
+
+class ClassKeyValue:
+    def __init__(self, key: str, value) -> None:
+        self.key = key
+        self.value = value
+        self.hash = hash(key)
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.length = 0
+        self.key_value_table: list = [None] * 8
+        # self.index = 0
+
+    def __len__(self) -> int:
+        return self.length
+
+    def __setitem__(self, key: str, value) -> None:
+        if self.length / len(self.key_value_table) >= 0.75:
+            self.double_size()
+        index = self.get_index(key)
+
+
+        for _ in range(len(self.key_value_table)):
+            key_value = self.key_value_table[index]
+            if key_value is None:
+                self.key_value_table[index] = ClassKeyValue(key, value)
+                self.length += 1
+                return
+            if self.key_value_table[index].key == key:
+                self.key_value_table[index].value = value
+                return
+
+            index = (index + 1) % len(self.key_value_table)
+
+    def __getitem__(self, key: str):
+        index = self.get_index(key)
+
+        # if key_value.key == key:
+        #     return key_value.value
+        for _ in range(len(self.key_value_table)):
+            key_value = self.key_value_table[index]
+            if key_value is not None and key_value.key == key:
+                return key_value.value
+            index = (index + 1) % len(self.key_value_table)
+
+        raise KeyError
```

Comment:
When raising a `KeyError`, it's a good practice to include the key that caused the error. This provides more context and helps with debugging. Consider changing `raise KeyError` to `raise KeyError(key)`.